modules/internal/ChapelLocale.chpl

```diff
@@ -153,71 +162,150 @@ module ChapelLocale {
+      the program has access to or do its best to ignore such limits.
+      By default it returns the number of accessible physical cores.
+
+      :arg logical: Count logical PUs (hyperthreads and the like),
+                    or physical ones (cores)?  Defaults to `false`,
+                    for cores.
+      :type logical: `bool`
+      :arg accessible: Count only PUs that can be reached, or all of
+                       them?  Defaults to `true`, for accessible PUs.
+      :type accessible: `bool`
+      :returns: number of PUs
+      :rtype: `int`
+
+      There are several things that can cause the OS to limit the
+      processor resources available to a Chapel program.  On plain
+      Linux systems using the ``taskset(1)`` command will do it.  On
+      Cray systems the ``CHPL_LAUNCHER_CORES_PER_LOCALE`` environment
+      variable may do it, indirectly via the system job launcher.
+      Also on Cray systems, using a system job launcher (``aprun`` or
+      ``slurm``) to run a Chapel program manually may do it, as can
+      running programs within Cray batch jobs that have been set up
+      with limited processor resources.
+     */
     inline proc numPUs(logical: bool = false, accessible: bool = true) {
       return this._value.numPUs(logical, accessible);
     }
 
+    /*
+      Get the integer identifier for this locale.
+
+      :returns: locale number, in the range ``0..numLocales-1``
+      :rtype: int
+     */
     inline proc id {
       return this._value.id;
     }
 
+    pragma "no doc"
     inline proc localeid {
       return this._value.localeid;
     }
 
+    /*
+      Get the hostname of this locale.
+
+      :returns: the hostname of the compute node associated with the locale
+      :rtype: string
+    */
     inline proc hostname {
       return this._value.hostname;
     }
 
+    /*
+      Get the name of this locale.  In practice, this is often the
+      same as the hostname, though in some cases (like when using
+      local launchers), it may be modified.
+
+      :returns: locale name
+      :rtype: string
+     */
     inline proc name {
       return this._value.name;
     }
 
+    pragma "no doc"
     inline proc chpl_id() {
       return this._value.chpl_id();
     }
 
+    pragma "no doc"
     inline proc chpl_localeid() {
       return this._value.chpl_localeid();
     }
 
+    pragma "no doc"
     inline proc chpl_name() {
       return this._value.chpl_name();
     }
 
+    pragma "no doc"
     inline proc defaultMemory() {
       return this._value.defaultMemory();
     }
 
+    pragma "no doc"
     inline proc largeMemory() {
       return this._value.largeMemory();
     }
 
+    pragma "no doc"
     inline proc lowLatencyMemory() {
       return this._value.lowLatencyMemory();
     }
 
+    pragma "no doc"
     inline proc highBandwidthMemory() {
       return this._value.highBandwidthMemory();
     }
 
+    pragma "no doc"
```

Comment:
I think this one could be documented, and is sensical with the NUMA locale model, for example.  I'd simply say something like "return the number of sub-locales in this locale."  At the same time, I think it's fine not to document it.